### PR TITLE
integrate keyCLI in eth

### PR DESCRIPTION
**Pull request by yann300**
All options used in ethkey should be available using eth options: 
eth account
eth wallet

### CODE REVIEW

File: eth/main.cpp [hunk 1]
```diff
@@ -425,6 +426,10 @@ int main(int argc, char** argv)
 		catch (...) {}
 	}
 
+	AccountManager accountm;
+	if (accountm.execute(argc, argv))
```
Comment (chriseth) [inline eth/main.cpp]: I think it is saner to add the "fork" here, i.e. let main.cpp decide when to forward execution to AccountManager.

File: eth/AccountManager.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * @author Yann <yann@ethdev.com>
+ * @date 2016
+ */
+#pragma once
+#include <libethcore/KeyManager.h>
+
+class AccountManager
+{
+public:
+	AccountManager() = default;
```
Comment (chriseth) [inline eth/AccountManager.h]: Not needed.

File: eth/AccountManager.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+#include "AccountManager.h"
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+bool AccountManager::execute(int argc, char** argv)
+{
+	if ((string)argv[1] == "wallet")
+	{
+		if (4 < argc && (string)argv[2] == "import")
```
Comment (chriseth) [inline eth/AccountManager.cpp]: Please don't use c-style casts

File: eth/AccountManager.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file AccountManager.h
+ * @author Yann <yann@ethdev.com>
+ * @date 2016
+ */
+#pragma once
+#include <libethcore/KeyManager.h>
+
+class AccountManager
```
Comment (chriseth) [inline eth/AccountManager.h:32]: Please add docstrings for most of the functions here.

File: eth/main.cpp [hunk 5]
```diff
@@ -425,6 +426,10 @@ int main(int argc, char** argv)
 		catch (...) {}
 	}
 
+	AccountManager accountm;
```
Comment (chriseth) [inline eth/main.cpp]: `eth --help` should also display the account and wallet options

File: eth/AccountManager.cpp [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+	}
+	return *m_keyManager;
+}
+
+void AccountManager::openWallet()
+{
+	string password;
+	while (true)
+	{
+		if (m_keyManager->load(password))
```
Comment (chriseth) [inline eth/AccountManager.cpp]: Why not use `keyManager()` here?

File: eth/AccountManager.cpp [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+		ret = getPassword(_prompt);
+		string confirm = getPassword("Please confirm the passphrase by entering it again: ");
+		if (ret == confirm)
+			break;
+		cout << "Passwords were different. Try again." << endl;
+	}
+	return ret;
+}
+
+pair<string, string> AccountManager::createPassword(std::string const& _prompt, std::string const& _pass, std::string const& _hint) const
```
Comment (chriseth) [inline eth/AccountManager.cpp]: No `std::`

**Comment by chriseth** [comment]
It seems that mist searches for specific texts in the output: https://github.com/ethereum/mist/blob/wallet/modules/ipcCommunicator.js#L63-L107

So on password failure, we should write `Decryption failed:` and on successful import, we should write `Address: {234567654321234567654323456}`

File: eth/AccountManager.cpp [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+bool AccountManager::execute(int argc, char** argv)
+{
+	if ((string)argv[1] == "wallet")
+	{
+		if (4 < argc && (string)argv[2] == "import")
+		{
+			keyManager();
+			string file =  argv[3];
+			string name = argv[4];
```
Comment (chriseth) [inline eth/AccountManager.cpp]: We cannot require a name here. If it does not work without name, can you try importing it as a bare key?

File: eth/AccountManager.cpp [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+
+		m_keyManager.reset();
+		return true;
+	}
+	else if ((string)argv[1] == "account")
+	{
+		if (2 < argc && (string)argv[2] == "list")
+		{
+			keyManager();
+			if (m_keyManager->store().keys().empty())
```
Comment (chriseth) [inline eth/AccountManager.cpp:85]: Why not just `if (keyManager().store().keys().empty())`?

File: eth/AccountManager.cpp [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+
+KeyManager& AccountManager::keyManager()
+{
+	if (!m_keyManager)
+	{
+		m_keyManager.reset(new KeyManager());
+		if (m_keyManager->exists())
+			openWallet();
+		else
+			cerr << "Couldn't open wallet. Does it exist?" << endl;
```
Comment (chriseth) [inline eth/AccountManager.cpp]: Shouldn't we exit here?

File: eth/AccountManager.cpp [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+		if (2 < argc && (string)argv[2] == "list")
+		{
+			keyManager();
+			if (m_keyManager->store().keys().empty())
+				cout << "No keys found." << endl;
+			else
+			{
+				vector<u128> bare;
+				AddressHash got;
+				for (auto const& u: m_keyManager->store().keys())
```
Comment (chriseth) [inline eth/AccountManager.cpp:92]: Would be great if this routine could work without a key manager, i.e. without master passphrase.

File: eth/AccountManager.cpp (hunk 8 above)
Comment (yann300) [inline eth/AccountManager.cpp]: if the keyManager is instanciated but the wallet never opened (using the Master password), everything will be read and imported as bare keys.
So we could never open the wallet if we want to be closer than how geth manage keys.

File: eth/AccountManager.cpp (hunk 10 above)
Comment (yann300) [inline eth/AccountManager.cpp]: if the wallet is not opened, accounts will still be listed as bare keys

**Comment by chriseth** [comment]
Build failes.

**Pull request merged**